src/colmap/scene/reconstruction_io.cc

```diff
@@ -481,10 +684,23 @@ void ReadImagesBinary(Reconstruction& reconstruction, std::istream& stream) {
     cam_from_world.translation.x() = ReadBinaryLittleEndian<double>(&stream);
     cam_from_world.translation.y() = ReadBinaryLittleEndian<double>(&stream);
     cam_from_world.translation.z() = ReadBinaryLittleEndian<double>(&stream);
-    image.SetCamFromWorld(cam_from_world);
 
     image.SetCameraId(ReadBinaryLittleEndian<camera_t>(&stream));
 
+    if (is_legacy_reconstruction) {
+      CreateFrameForImage(reconstruction, image);
+      image.SetFrameId(image.ImageId());
+      image.SetFramePtr(&reconstruction.Frame(image.ImageId()));
+    } else {
+      Frame* frame = image_to_frame.at(image.ImageId());
+      image.SetFrameId(frame->FrameId());
+      image.SetFramePtr(frame);
+    }
+
+    if (image.HasTrivialFrame()) {
+      image.FramePtr()->SetFrameFromWorld(cam_from_world);
```

Comment:
Do I understand correctly that each image pose is now written both to the images and to the frames files? And that it is again read twice? Is this better than incrementing the file version (is there one?) and storing the pose in the frames file only?

Reply:
We don't have a file version unfortunately. I did this to be fully forwards and backwards compatible. Anybody that deserializes the image poses (e.g., for gaussian splatting), can read the files with the exact same code and get valid image poses. Eventually, we should move to a format versioning and/or protobuf. Since the pose data is so small compared to the keypoints, tracks, etc. I am not worried about the duplication.